pkg/policy/mapstate.go

```diff
@@ -220,32 +244,176 @@ func getNets(identities Identities, ident uint32) []*net.IPNet {
 	return identities.GetNetsLocked(id)
 }
 
+// NewMapState creates a new MapState interface
+func NewMapState(initMap map[Key]MapStateEntry) MapState {
+	return newMapState(initMap)
+}
+
+func newMapState(initMap map[Key]MapStateEntry) *mapState {
+	m := &mapState{
+		allows: make(map[Key]MapStateEntry),
+		denies: make(map[Key]MapStateEntry),
+	}
+	for k, v := range initMap {
+		m.Insert(k, v)
+	}
+	return m
+}
+
+// Get the MapStateEntry that matches the Key.
+func (ms *mapState) Get(k Key) (MapStateEntry, bool) {
+	if ms == nil || (ms.allows == nil && ms.denies == nil) {
+		return MapStateEntry{}, false
+	}
+	if ms.denies == nil {
```

Comment:
When would these be nil? I would just simplify this down to first checking denies and then checking allows.